csrc/evaluator_common.cpp

```diff
@@ -435,14 +443,49 @@ void NaiveValueMachine::makeBinaryOp(BinaryOp* bop) {
   dest_[index] = out;
 }
 
+void NaiveValueMachine::makeTernaryOp(TernaryOp* top) {
+  int in0 = top->inputs()[0]->evaluatorIndex();
+  int in1 = top->inputs()[1]->evaluatorIndex();
+  int in2 = top->inputs()[2]->evaluatorIndex();
+  int out = top->outputs()[0]->evaluatorIndex();
+
+  NVF_ERROR(in0 >= 0, "Integer Machine: unknown first input: ", top);
+  NVF_ERROR(in1 >= 0, "Integer Machine: unknown second input: ", top);
+  NVF_ERROR(in2 >= 0, "Integer Machine: unknown third input: ", top);
+  NVF_ERROR(out >= 0, "Integer Machine: unknown out: ", top);
+
+  int index = makeInstructionEntry();
+  inst_type_[index] = InstructionType::TERNARY_OP;
+  top_type_[index] = top->getTernaryOpType();
+  src0_[index] = in0;
+  src1_[index] = in1;
+  src2_[index] = in2;
+  dest_[index] = out;
+}
+
+void NaiveValueMachine::makeSetOp(LoadStoreOp* lsop) {
+  int in = lsop->in()->evaluatorIndex();
+  int out = lsop->out()->evaluatorIndex();
+
+  NVF_ERROR(in >= 0, "Integer Machine: unknown input: ", lsop);
+  NVF_ERROR(out >= 0, "Integer Machine: unknown out: ", lsop);
+
+  int index = makeInstructionEntry();
+  inst_type_[index] = InstructionType::SET_OP;
+  src0_[index] = in;
+  dest_[index] = out;
+}
+
```

Comment:
I can make this a separate PR also, but it's very small since half of the machinery for `set` was already in place.